spec/requests/api/rest/admin/gateways_spec.rb

```diff
@@ -433,4 +526,101 @@
       end
     end
   end
+
+  describe 'GET /api/rest/admin/gateways/{id}' do
+    subject do
+      get json_api_request_path, params: request_params, headers: json_api_request_headers
+    end
+
+    before do
+      gateway.save!
+    end
+
+    let(:json_api_request_path) { "#{super()}/#{record_id}" }
+    let(:request_params) { nil }
+    let(:record_id) { gateway.id.to_s }
+
+    include_examples :returns_json_api_record, relationships: gateway_relationship_names do
+      let(:json_api_record_id) { record_id }
+      let(:json_api_record_attributes) { gateway_response_attributes }
+    end
+  end
+
+  describe 'POST /api/rest/admin/gateways' do
+    subject do
+      post json_api_request_path, params: json_api_request_body.to_json, headers: json_api_request_headers
+    end
+
+    let(:contractor) { FactoryBot.create(:customer) }
+    let(:session_refresh_method) { SessionRefreshMethod.first }
+    let(:sdp_alines_filter_type) { FilterType.first }
+    let(:codec_group) { FactoryBot.create(:codec_group) }
+    let(:sdp_c_location) { SdpCLocation.first }
+    let(:sensor) { FactoryBot.create(:sensor) }
+    let(:sensor_level) { System::SensorLevel.first }
+    let(:dtmf_receive_mode) { System::DtmfReceiveMode.first }
+    let(:dtmf_send_mode) { System::DtmfSendMode.first }
+    let(:transport_protocol) { Equipment::TransportProtocol.first }
+    let(:term_proxy_transport_protocol) { Equipment::TransportProtocol.first }
+    let(:orig_proxy_transport_protocol) { Equipment::TransportProtocol.first }
+    let(:rel100_mode) { Equipment::GatewayRel100Mode.first }
+    let(:rx_inband_dtmf_filtering_mode) { Equipment::GatewayInbandDtmfFilteringMode.first }
+    let(:tx_inband_dtmf_filtering_mode) { Equipment::GatewayInbandDtmfFilteringMode.first }
+    let(:network_protocol_priority) { Equipment::GatewayNetworkProtocolPriority.first }
+    let(:media_encryption_mode) { Equipment::GatewayMediaEncryptionMode.first }
+    let(:sip_schema) { System::SipSchema.first }
+
+    let(:json_api_request_attributes) do
+      {
+        'name': gateway.name,
+        'enabled': gateway.enabled,
+        'priority': gateway.priority,
+        'weight': gateway.weight,
+        'acd-limit': gateway.acd_limit,
+        'asr-limit': gateway.asr_limit,
+        'host': gateway.host
+      }
+    end
```

Comment:
use actual valus in hash, like
```ruby
    let(:json_api_request_attributes) do
      {
        name: 'test gateway',
        enabled: false,
        # ....
      }
    end
```

also no need to write symbol in single quote if it has only letters/digits

so `{ 'host': 'test.com' }` and `:'host'` should be written like  `{ host: 'test.com' }` and `:host`

but `{ 'asr-limit': 123 }` and `:'asr-limit'` can be written only in quotes

Reply:
same for other hashes in this file